UNLV-CS472-672/2024-S-GROUP5-Munch
# Creating Comment API Routes and Helper Function for Generating unique IDs

## Description / Changes Made

- Created comment API route, including a patch method for when a comment is added to the array of comments, as well as a delete method.
- Created a helper function that will generate a unique ID, this is used in the comment by looking for a post, then being able to detect the comment ID that the user is requesting to delete.
- Takes in the user's ID, post's ID, as well as if the comment is being deleted then you will also be required the comment ID
- When generating the new comment, the request takes in a JSON that only requires the "comment" field, which will update Firebase DB comment array with that comment alongside an author, comment_id, creation_date, and their username.

## Portions of this code that utilized AI generation

- Link to Generated Code (If Applicable): https://chat.openai.com/share/0b43a2b9-8681-4026-9516-106f76f2b73d

AI was used to find a method to create proper formatting for current time that is utilized in creation of the comment, as well as helping create the helper function that generates a unique ID!

## How to Test

(Assuming Postman is being used)

1. Use endpoint including "/api/posts/comment/<user_id>/<post_id>" to create a new comment
2. Go to body tab, then click raw, then insert JSON information. For this test, you can just use {"comment": "Any Comment"} as an example
3. Make sure Postman is set to PATCH request and send request.
4. If using delete functionality, endpoint would be "/api/posts/comment/<user_id>/<post_id>/<comment_id>"
5. Make sure Postman is set to DELETE request and send request.

## Checklist
